src/Engine/ProtoCore/FFI/CLRDLLModule.cs

```diff
@@ -782,7 +782,10 @@ private void RegisterFunctionPointer(string functionName, MemberInfo method, Lis
             if (CoreUtils.IsDisposeMethod(functionName))
                 f = new DisposeFunctionPointer(Module, method, retype);
             else if (CoreUtils.IsGetter(functionName))
+            {
                 f = new GetterFunctionPointer(Module, functionName, method, retype);
+                (f as GetterFunctionPointer).ReflectionInfo.CheckForRankReductionAttribute(mGetterAttributes);
```

Comment:
Utilizing ReflectionInfo is usually performance sensitive especially under a for loop, can you give us your estimation of the performance impact? Is it highly depending on how many getters there are?

Reply:
I created a test zero touch class with 100 properties with attributes and the call to `CheckForRankReductionAttribute()` took <1ms.